### ISSUE TITLE
"different" validation is not compatible with "nullable" validation

### ISSUE THREAD

**Issue by robjbrain** [laravel/framework#26749]
- Laravel Version: 5.7.15
- PHP Version: 7.2

### Description:
The validation for `different` is not compatible with the validation for `nullable` since even if you define a field as `nullable` it will fail the difference check if the "other" value is null.

What's more whether or not it passes or fails is dependent on which way round you order the rules.

If you have two values "old_value = null" and "new_value = 123" then it will fail if `different` is a rule on "old_value" but not if `different` is a rule on "new_value" even though you would expect the result to be the same because you're comparing the same two values.

But if you put the different rule on 

### Steps To Reproduce:
```
$validator = Validator::make([
        'old_value' => 123,
        'new_value' => null
    ], [
        'old_value' => ['present', 'nullable', 'different:new_value'],
        'new_value' => ['present', 'nullable'],
    ]);
dump($validator->errors());
```

> {"old_value":["The old value and new value must be different."]}

```
$validator = Validator::make([
        'old_value' => 123,
        'new_value' => null
    ], [
        'old_value' => ['present', 'nullable'],
        'new_value' => ['present', 'nullable', 'different:old_value'],
    ]);
dump($validator->errors());
```

> []

```
$validator = Validator::make([
        'old_value' => null,
        'new_value' => 123
    ], [
        'old_value' => ['present', 'nullable'],
        'new_value' => ['present', 'nullable', 'different:old_value'],
    ]);
dump($validator->errors());
```

> {"new_value":["The new value and old value must be different."]}

```
$validator = Validator::make([
        'old_value' => null,
        'new_value' => 123
    ], [
        'old_value' => ['present', 'nullable', 'different:new_value'],
        'new_value' => ['present', 'nullable'],
    ]);
dump($validator->errors());
```

> []

### Expected Behaviour:
`different` should simple compute whether the two values are different or the same and not additionally fail if the value is null. Whether or not it can be null should be left up to other validation rules such as `required` and `nullable`.

**Comment by staudenmeir** [comment]
The validator explicitely checks whether the other value is not `null`:

https://github.com/laravel/framework/blob/5f6d9a431c4fcedcbae969ba95d266604be54060/src/Illuminate/Validation/Concerns/ValidatesAttributes.php#L432-L445

That's why `different:old_value` works but `different:new_value` doesn't.

I assume this is done to fail the validation when the other value doesn't exist:

https://github.com/laravel/framework/blob/5f6d9a431c4fcedcbae969ba95d266604be54060/tests/Validation/ValidationValidatorTest.php#L1050-L1051

We could fix it by removing `is_null($other)` and using `Arr::has()` instead.

**Comment by staudenmeir** [comment]
Will be fixed in Laravel 5.8.

**Comment by jcfrazier6** [comment]
This is still happening in Laravel 8. Different and Nullable aren't able to be used together. Both fields have the same value. Different only works if nullable isn't used.
